src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/error/UnexpectedInputException.java

```diff
@@ -0,0 +1,36 @@
+/* Copyright 2022 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.error;
+
+import org.openkilda.messaging.Message;
+import org.openkilda.messaging.MessageCookie;
+import org.openkilda.messaging.MessageData;
+
+public class UnexpectedInputException extends RuntimeException {
+    private final Object input;
+
+    public UnexpectedInputException(Message input) {
+        this.input = input;
+    }
+
+    public UnexpectedInputException(MessageData payload) {
+        this.input = payload;
+    }
+
+    public String getMessage(MessageCookie rootCookie) {
```

Comment:
I'd rather expect this cookie is taken from inputs payload.

Reply:
There is no cookie in input, if it is `MessageData` (and the cookie available toogether with this `MessageData` will be unwrapped on some set of layers and as result will not able to provide all required info for error message). 

But dispatching method must have "root" (from it's perspective) cookie object and can provide it for error message construction.